firmware/cmake/embedded.cmake

```diff
@@ -40,7 +40,6 @@ set(SHARED_COMPILER_FLAGS
         --specs=nosys.specs
         --specs=nano.specs
         -Wall
-        -Werror
```

Comment:
Let's add back `-Werror`, I think this is a consequence of our merges - I deleted it way back but it should be there.